### PR TITLE
On Azure .exists blob timeout, log the exception and return False

### PR DESCRIPTION
**Pull request by humitos**
We have been experimenting some issues with Azure Blob storage. Our current timeout is the django-storages default (20 secs). We want to reduce it to ~2s, but without breaking the code and start returning 500. So, we log the error and return False.

### CODE REVIEW

**Comment by humitos** [comment]
I'm not sure that this really works. I wasn't able to reproduce it locally with lower azure timeout, or even stopping the storage container.

File: readthedocs/storage/azure_storage.py [hunk 1]
```diff
@@ -30,6 +34,14 @@ def url(self, name, expire=None, http_method=None):  # noqa
         """
         return super().url(name, expire)
 
+    def exists(self, name, timeout=None):
```
Comment (ericholscher) [inline readthedocs/storage/azure_storage.py:37]: We don't actually use the `timeout` here, so probably shouldn't accept it.

Comment (humitos) [reply readthedocs/storage/azure_storage.py:37]: Meh! I was working in the other idea about passing a timeout here and forgot to remove it :/

**Comment by humitos** [comment]
Yeah. Feel the same. I don't have another idea about how to test it, but I don't understand why it does not work when stopping the storage container either. The custom `.exists` function is executed (I put a pdb there and it stopped), but the timeout seems to not be respected.

I think _the code_ won't fail, but it may just does not do what we expect it to do :confused:

**Comment by ericholscher** [comment]
@humitos I'm happy to merge it once we fix the tests, and we can see what happens in prod.

**Pull request merged**